### ISSUE TITLE
hankel edge cases and documentation issues

### ISSUE THREAD

**Issue by kostrzewa** [HISKP-LQCD/hadron#224]
Looking through `hankel.R` (which was prematurely deployed in my opinion), I can see a number of potential edge cases which are not documented, nor taken into account. Please let me know if I'm mistaken.

For example, the fact that (a *single*) `NA` is returned in `gevp.hankel` when the QR decomposition fails to provide coefficients should be documented. What if I had specified `id=c(1,2)` with the intent of getting the first two eigenvalues?

Another edge case is what happens when: 

```
  ii <- which(abs(Im(M.eigen$values)) < eps & Re(M.eigen$values) > range[1]
              & Re(M.eigen$values) < range[2])
```

comes out empty?

The existence of `id="all"` is not documented and not supported via `bootstrap.hankel` as `length(id)` is used in multiple places.

Clarification is required for `range`, it should be documented that this restricts the possibly extracted eigenvalues, just like `eps` does.

In the description of `bootstrap.hankel`, the mention of `correlation matrices` is kind of misleading at present as the function currently only operates on a single correlator.

The step `M[abs(M] < 1.e-12]` with a magic number is not commented on. Why `1.e-12` and not some other value?

The return value of `bootstrap.hankel` should be documented.

In `gevp.hankel`, `cf` is a vector and yet `gevp.hankel` inherits the documentation of `bootstrap.hankel` which specifies that `cf` is an object of class `cf`.

In `hankel2effectivemass`, `ret$cf <- cf` makes no sense, `cf` is a function.

In `hankel2effectivemass`, supported effmass types should be documented.

**Comment by kostrzewa** [comment]
In `hankel2cf`, the value of `endcorr` used for `cf_boot` should come from `hankel$cf$endcorr`.

**Comment by martin-ueding** [comment]
> which was prematurely deployed in my opinion

I thought of making the current state available as it does not seem to break the other stuff. This way I can use the `master` in the analysis. It seems that we are working slightly different and perhaps we would need to introduce say git-flow to the hadron project as many more people become involved. Though I see the issue that git-flow will not exactly the people who struggle with git already.

**Comment by urbach** [comment]
> > which was prematurely deployed in my opinion
> 
> I thought of making the current state available as it does not seem
> to break the other stuff. This way I can use the `master` in the
> analysis. It seems that we are working slightly different and
> perhaps we would need to introduce say git-flow to the hadron
> project as many more people become involved. Though I see the issue
> that git-flow will not exactly the people who struggle with git
> already. 

Well, we should work on the remaining points raised. I think in this
case the merged code is already much better than what was there
before. And in its current state it is pretty useful for our ongoing
analyses. I'll take care of the remaining points, there is anyhow more
to take care of and to understand.

**Comment by urbach** [comment]
see #229, not finished yet.

**Comment by kostrzewa** [comment]
This was resolved by #229
